src/Symfony/Component/FeatureFlag/FeatureChecker.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureFlag;
+
+final class FeatureChecker implements FeatureCheckerInterface
+{
+    private array $cache = [];
+
+    public function __construct(
+        private readonly FeatureRegistryInterface $featureRegistry,
+        private readonly mixed $default = false,
+    ) {
+    }
+
+    public function isEnabled(string $featureName, mixed $expectedValue = true): bool
+    {
+        return $this->getValue($featureName) === $expectedValue;
+    }
+
+    public function getValue(string $featureName): mixed
+    {
+        try {
+            return $this->cache[$featureName] ??= $this->featureRegistry->get($featureName)();
+        } catch (FeatureNotFoundException) {
+            return $this->default;
```

Comment:
Having a common default value for all kinds of flags looks wrong to me. Shouldn't it be an argument of the current method itself?

Reply:
IMO, this will make FeatureChecker harder to use due to a new argument in `isEnabled` and `getValue` methods and harder to set up a global default (which might be the most common use case).

Reply:
The point is that currently there is no way to set up a custom default value for one specific feature, which is a common case too.

That makes me think that we should improve the design in order to create multiple checkers instead of only one, or the feature itself should be aware about its default value.

Reply:
> The point is that currently there is no way to set up a custom default value for one specific feature, which is a common case too.

Do you have in mind a use case like the following ?
```
$featureChecker->getValue('code-version'); // get version or fallback to the FeatureChecker's default value
$featureChecker->getValue('code-version', 'v1'); // get version or fallback to 'v1'
```
In this case, we will have to add a `mixed $default` to both `isEnabled` and `getValue` methods. Depending on the number of arguments (1 or 2), it will fallback to the global default value. 
=> The default value will be defined per call.

We can also create a ChainFeatureRegistry which will ask inner registries for the feature's value : the initial `FeatureRegistry` and a `DefaultFeatureRegistry` which contain a map of default value per feature. 
=> The default value will be defined per feature.

> That makes me think that we should improve the design in order to create multiple checkers instead of only one.

For now, we already can create multiple FeatureChecker instances. They will only differ in their feature list and the global default value.

> or the feature itself should be aware about its default value

The default is used when the feature is not found. How can we set a default value for a non-existent feature?

Reply:
I've another concern related to this fallback behavior; it occurs only when the feature does not exist (still, you might want to set up a different default depending on the feature's type). However, in most cases the app is aware of the feature flags available as they are hardcoded using their names. So a "feature not found" could also mean a logic error caused by a developer's typo or outdated code, resulting in silent failure; and you won't know if the feature exists or not, as a valid value will always be returned as a result.

I'm now wondering what the better approach is: should we throw the exception or return a fallback value? Imho, the component implementation should throw the exception.

Reply:
>We can also create a ChainFeatureRegistry which will ask inner registries for the feature's value : the initial FeatureRegistry and a DefaultFeatureRegistry which contain a map of default value per feature.
=> The default value will be defined per feature.

`$default`-apart, I support that idea, it will enable multiple FF providers if desired.